sstable/block.go

```diff
@@ -693,6 +704,17 @@ func (i *blockIter) SeekGE(key []byte, flags base.SeekGEFlags) (*InternalKey, ba
 		panic(errors.AssertionFailedf("invalidated blockIter used"))
 	}
 
+	searchKey := key
```

Comment:
There are a few places further down in this function which use `key` instead of `searchKey`, and I'm not sure it's correct. One [example](https://github.com/cockroachdb/pebble/blob/97ec6a5118e3c25368a963054ed0839a244178f9/sstable/block.go#L830).  To prevent this sort of error, perhaps we don't need to introduce `searchKey` as a new variable?

Reply:
> To prevent this sort of error, perhaps we don't need to introduce searchKey as a new variable?
I revoke this suggestion. We don't want to change the where the client provided `key` points to.